Tests.Integration.Presentation.Web/Advice/AdviceTest.cs

```diff
@@ -184,36 +131,136 @@ public async Task Cannot_Add_Repeatable_Advice_With_StopDate_Before_StartDate()
+            //Arrange
+
+            //Act
+            var deleteResult = await AdviceHelper.DeleteAdviceAsync(0);
+
+            //Assert
+            Assert.Equal(HttpStatusCode.NotFound, deleteResult.StatusCode);
+        }
+
+        [Fact]
+        public async Task Cannot_Delete_Advice_That_Has_Been_Sent()
+        {
+            //Arrange
+            var recipient = createDefaultEmailRecipient(createWellformedEmail());
+            var createAdvice = CreateDefaultAdvice(Scheduling.Day, AdviceType.Immediate, recipient);
+
+            var createResult = await AdviceHelper.PostAdviceAsync(createAdvice, OrganizationId);
+            Assert.Equal(HttpStatusCode.Created, createResult.StatusCode);
+            var createdAdvice = await createResult.ReadResponseBodyAsAsync<Core.DomainModel.Advice.Advice>();
+
+            Thread.Sleep(3000); //Sleep for 3 seconds to allow the immediate advis to be sent.
+
+            //Act
+            var deleteResult = await AdviceHelper.DeleteAdviceAsync(createdAdvice.Id);
+
+            //Assert
+            Assert.Equal(HttpStatusCode.BadRequest, deleteResult.StatusCode);
+        }
+
+        [Fact]
+        public async Task Cannot_Delete_Inactive_Advice_That_Has_Not_Been_Sent_If_No_Rights()
+        {
+            //Arrange
+            var recipient = createDefaultEmailRecipient(createWellformedEmail());
+            var createAdvice = CreateDefaultAdvice(Scheduling.Day, AdviceType.Repeat, recipient);
+
+            var createResult = await AdviceHelper.PostAdviceAsync(createAdvice, OrganizationId);
+            Assert.Equal(HttpStatusCode.Created, createResult.StatusCode);
+            var createdAdvice = await createResult.ReadResponseBodyAsAsync<Core.DomainModel.Advice.Advice>();
+
+            var deactivateResult = await AdviceHelper.DeactivateAdviceAsync(createdAdvice.Id);
+            Assert.Equal(HttpStatusCode.NoContent, deactivateResult.StatusCode);
+
+            var regularUserCookie = await HttpApi.GetCookieAsync(OrganizationRole.User);
+
+            //Act
+            var deleteResult = await AdviceHelper.DeleteAdviceAsync(createdAdvice.Id, regularUserCookie);
+
+            //Assert
+            Assert.Equal(HttpStatusCode.Forbidden, deleteResult.StatusCode);
+        }
+
+        private Core.DomainModel.Advice.Advice CreateDefaultAdvice(Scheduling schedule, AdviceType type, AdviceUserRelation recipient)
+        {
+            return new Core.DomainModel.Advice.Advice
             {
                 Body = A<string>(),
                 Subject = A<string>(),
-                Scheduling = Core.DomainModel.Advice.Scheduling.Day,
+                Scheduling = Scheduling.Day,
                 AdviceType = AdviceType.Repeat,
-                Reciepients = new List<Core.DomainModel.Advice.AdviceUserRelation>()
+                Reciepients = new List<AdviceUserRelation>()
                 {
                     recipient
                 },
                 RelationId = A<int>(),
```

Comment:
OBS: Relation Id skal pege på et objekt og der skal være en type på. Det er det jeg retter nu. Du kommer ikke til at kunne oprette en advis uden type og relation id for det er root den adgangstjekker